_ingest-pipelines/processors/dot-expander.md

```diff
@@ -0,0 +1,315 @@
+  "processors": [
+    {
+      "dot_expander": {
+        "field": "user.address.city"
+      }
+    },
+    {
+        "dot_expander":{
+         "field": "user.address.state"
+        }
+    }
+  ]
+}
+```
+{% include copy-curl.html %}
+
+**Step 2 (Optional): Test the pipeline.**
+
+It is recommended that you test your pipeline before you ingest documents.
+{: .tip}
+
+To test the pipeline, run the following query:
+
+```json
+POST _ingest/pipeline/dot-expander/_simulate
+{
+  "docs": [
+    {
+      "_index": "testindex1",
+      "_id": "1",
+      "_source": {
+        "field": "city, state"
+      }
+    }
+  ]
+}
+```
+{% include copy-curl.html %}
+
+#### Response
+
+The following example response confirms that the pipeline is working as expected:
+
+```json
+{
+  "docs": [
+    {
+      "doc": {
+        "_index": "testindex1",
+        "_id": "1",
+        "_source": {
+          "field": "city, state"
+        },
+        "_ingest": {
+          "timestamp": "2023-11-17T23:49:27.597933805Z"
+        }
+      }
+    }
+  ]
+}
+```
+
+**Step 3: Ingest a document.**
+
+The following query ingests a document into an index named `testindex1`:
+
+```json
+PUT testindex1/_doc/1?pipeline=dot-expander
+{
+  "field": "Denver, CO"
+}
+```
+{% include copy-curl.html %}
+
+**Step 4 (Optional): Retrieve the document.**
+
+To retrieve the document, run the following query:
+
+```json
+GET testindex1/_doc/1
+```
+{% include copy-curl.html %}
+
+#### Response
+
+The following response confirms the document was indexed:
+
+```json
+{
+  "_index": "testindex1",
+  "_id": "1",
+  "_version": 58,
+  "_seq_no": 57,
+  "_primary_term": 30,
+  "found": true,
+  "_source": {
+    "field": "Denver, CO"
+  }
+}
+```
+
+## Nested fields
+
+The processor consolidates the `user.address.city` and `user.address.state` fields by merging with an existing `address`, `city`, and `state` field nested under `user`. If the field is a scalar value, then it will turn that into an array field. Take for example the following document:
+
+```json
+{
+    "user": {
+        "address": {
+            "city": "Denver",
+            "state": "CO"
+        }
+    }
+}
+```
+
+The `dot_expander` processor transforms the individual fields into arrays as follows:
+
+```json
+{
+    "user": {
+        "address": {
+            "city": ["Denver"],
+            "state": ["CO"]
+        }
+    }
+}
+```
+
+If you set the `override` parameter to `true`, the value of the expanded field overrides the value of the nested object. Take for example the following configuration:
+
+```json
+{
+    "dot_expander": {
+        "field": "user.address.city",
+        "override": true
+    }
+}
+```
+
+The result is the following document, in which the expanded field `user.address.city` overrides the value of the nested object `user.address`:
+
+```json
+{
+    "user": {
+        "address": {
+            "city": "Denver",
+            "state": "CO"
+        }
+    }
+}
+```
```

Comment:
A better example would be an input like:

```
{
  "user.address.city": "Boulder",
   "user": {
        "address": {
            "city": "Denver",
            "state": "CO"
        }
    }
}
```

With `override:false` (or unset), the dot-pathed top-level field gets appended to the target object node, so it transforms into:

```
{
   "user": {
        "address": {
            "city": ["Denver", "Boulder"],
            "state": "CO"
        }
    }
}
```

With `override:true`, the dot-pathed top-level field value overwrites the target object node, so it transforms into:

```
{
   "user": {
        "address": {
            "city": "Boulder",
            "state": "CO"
        }
    }
}
```